### ISSUE TITLE
Add a fit_transform() convenience method

### ISSUE THREAD

**Issue by perimosocordiae** [all-umass/metric-learn#25]
We're already following the scikit-learn method conventions, might as well make it easy to fit and transform in a single step.

**Comment by bhargavvader** [comment]
I'll try and take a shot at this!

**Comment by bhargavvader** [comment]
Can this issue be closed now?

**Comment by perimosocordiae** [comment]
Yep, thanks!
